lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -283,6 +257,70 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
         }
     }
 
+    @Tracing
+    private Map<String, Object> checkForCIScoringFailure(
+            String ipAddress,
+            ClientOAuthSessionItem clientOAuthSessionItem,
+            String govukSigninJourneyId)
+            throws CiRetrievalException, ConfigException {
+        var contraIndicators =
+                ciMitService.getContraIndicatorsVC(
+                        clientOAuthSessionItem.getUserId(), govukSigninJourneyId, ipAddress);
+
+        // CI scoring failure
+        if (ciMitUtilityService.isBreachingCiThreshold(contraIndicators)) {
+            return ciMitUtilityService
+                    .getCiMitigationJourneyStep(contraIndicators)
+                    .orElse(JOURNEY_FAIL_WITH_CI)
+                    .toObjectMap();
+        }
+        return new HashMap<>();
+    }
+
+    @Tracing
+    private Map<String, Object> checkForProfileMatchAndVCsCheck(
+            IpvSessionItem ipvSessionItem,
+            ClientOAuthSessionItem clientOAuthSessionItem,
+            AuditEventUser auditEventUser,
+            List<VcStoreItem> vcStoreItems,
+            boolean isF2FComplete)
+            throws ParseException, UnknownEvidenceTypeException, SqsException,
+                    HttpResponseExceptionWithErrorBody, CredentialParseException {
+        List<SignedJWT> credentials =
+                gpg45ProfileEvaluator.parseCredentials(
+                        userIdentityService.getIdentityCredentials(vcStoreItems));
+
+        // Check for attained vot from vtr
+        var strongestAttainedVotFromVtr =
+                getStrongestAttainedVotForVtr(
+                        clientOAuthSessionItem.getVtr(), credentials, vcStoreItems, auditEventUser);
+
+        // vot achieved for vtr
+        if (strongestAttainedVotFromVtr.isPresent()) {
+            // Profile matched
+            Vot attainedVot = strongestAttainedVotFromVtr.get();
+            ipvSessionItem.setVot(attainedVot.name());
+            ipvSessionService.updateIpvSession(ipvSessionItem);
+            if (attainedVot.getProfileType().equals(ProfileType.GPG45)
+                    && (!userIdentityService.areVCsCorrelated(vcStoreItems))) {
+                // Matched - Credential correlation failure
+                return isF2FComplete
+                        ? buildF2FNotCorrelatedResponse(auditEventUser)
+                        : buildNotCorrelatedResponse(auditEventUser);
+            }
```

Comment:
Sorry Rohit, I think this is still not quite in the right place... I think we need to include a correlation check in `achievedWithGpg45Profile`. In particular, if a user has a complete but uncorrelated set of VCs, they should still be able to use an operational profile.

This will make it hard to distinguish `buildF2FNotCorrelatedResponse` from `buildF2FNoMatchResponse`... They both result in the `f2f-fail` journey event, but record different audit events. Hopefully you can figure it out!

We might also mix up `buildNotCorrelatedResponse` and `buildNoMatchResponse`, but those record the same event, so not such a big problem.

Reply:
This was just refactoring to resolve sonar issue... yeah working on that.